### PR TITLE
Cache utils related cleanup

### PR DESCRIPTION
**Pull request by michaelbenayoun**
# What does this PR do?

Removes obsolete code related to the previous cache system for training.

### CODE REVIEW

**Review by JingyaHuang — APPROVED** [review]
LGTM, thanks for the housekeeping (˵ •̀ ᴗ - ˵ ) ✧

File: optimum/neuron/utils/hub_neuronx_cache.py [hunk 1]
```diff
@@ -238,23 +238,14 @@ def download_file_to_string(self, filename: str, limit: int = None):
...
-
 def create_hub_compile_cache_proxy(
     cache_url: Optional[CacheUrl] = None,
     cache_repo_id: Optional[str] = None,
 ):
     if cache_url is None:
         cache_url = CacheUrl.get_cache_url()
     if cache_repo_id is None:
-        cache_repo_id = get_hub_cache()
+        cache_repo_id = get_hf_hub_cache_repo()
```
Comment (dacorvo) [inline optimum/neuron/utils/hub_neuronx_cache.py:248]: I think there is a misunderstanding here: what is expected here is really a hub repo, not a local directory.
The local directory is the one provided by cacheurl.

Comment (dacorvo) [reply optimum/neuron/utils/hub_neuronx_cache.py:248]: The misunderstanding is on my side: could you add a comment explaining that there is a priority between the several hub cache repos ?

Comment (michaelbenayoun) [reply optimum/neuron/utils/hub_neuronx_cache.py:250]: Solved offline

**Review by dacorvo — APPROVED** [review]
LGTM, thanks !

**Comment by dacorvo** [comment]
Let's merge this after the release, shall we ?

**Comment by michaelbenayoun** [comment]
Merging before the release since we need the fix to be able to fetch from a remote cache repo.

**Pull request merged**